### PR TITLE
spotifyd: 0.3.5-unstable-2024-02-18 -> 0.3.5-unstable-2024-07-10; enable `withKeyring` by default; refactor

### PR DESCRIPTION
**Pull request by getchoo**
## Description of changes

Diff: https://github.com/Spotifyd/spotifyd/compare/ff2f7a06e54bf05afd57a0243dc9f67abc15f040..8fb0b9a5cce46d2e99e127881a04fb1986e58008

The highlight of this bump is the addition of a Jack backend on Linux. I have also refactored some things, added a basic test as we don't run checks, enabled keyring support by default, and added myself as a maintainer - let me know if any of these should be dropped :)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by JohnRTitor** [comment]
Result of `nixpkgs-review pr 326670` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>spotifyd</li>
  </ul>
</details>

File: pkgs/by-name/sp/spotifyd/package.nix [hunk 1]
```diff
@@ -15,7 +15,7 @@
   withPortAudio ? stdenv.isDarwin,
   portaudio,
   withMpris ? stdenv.isLinux,
-  withKeyring ? false,
+  withKeyring ? true,
```
Comment (JohnRTitor) [inline pkgs/by-name/sp/spotifyd/package.nix:18]: This can be `stdenv.isLinux` I believe

Comment (getchoo) [reply pkgs/by-name/sp/spotifyd/package.nix:18]: The keyring and it's [backend](https://github.com/hwchen/keyring-rs) are cross platform -- you can see them building with this in CI for MacOS [here](https://github.com/Spotifyd/spotifyd/blob/8fb0b9a5cce46d2e99e127881a04fb1986e58008/.github/workflows/ci.yml#L61-L62)

This did make me realize `dbus` will only need to be used on Linux, though :)

**Review by JohnRTitor — APPROVED** [review]
LGTM

**Pull request merged**